tools/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/GradleCore.java

```diff
@@ -221,6 +228,87 @@ public boolean visit( IResourceDelta delta ) throws CoreException
                 }
             }
         }, IResourceChangeEvent.POST_CHANGE );
+
+        CoreUtil.getWorkspace().addResourceChangeListener( new IResourceChangeListener()
+        {
+
+            @Override
+            public void resourceChanged( IResourceChangeEvent event )
+            {
+                try
+                {
+                    if( event.getType() == IResourceChangeEvent.PRE_DELETE )
+                    {
+                        // for the event of delete project
+                        IProject project = (IProject) event.getResource();
+                        if( LiferayWorkspaceUtil.isValidWorkspace( project ) )
+                        {
+                            IFolder bundlesFolder = project.getFolder(
+                                LiferayWorkspaceUtil.loadConfiguredHomeDir( project.getLocation().toOSString() ) );
+
+                            if( bundlesFolder.exists() )
+                            {
+                                File portalBundle = bundlesFolder.getLocation().toFile().getCanonicalFile();
+
+                                ServerUtil.deleteRuntimeAndServer( PortalRuntime.ID , portalBundle );
+                            }
+                        }
+
+                        return;
+                    }
+                    else
+                    {
+                        event.getDelta().accept( new IResourceDeltaVisitor()
+                        {
+
+                            @Override
+                            public boolean visit( IResourceDelta delta ) throws CoreException
+                            {
+                                // for only delete bundles dir
+                                if( delta.getKind() == IResourceDelta.REMOVED )
+                                {
+                                    IResource deletedRes = delta.getResource();
+
+                                    IProject project = deletedRes.getProject();
+
+                                    if( LiferayWorkspaceUtil.isValidWorkspace( project ) )
+                                    {
+                                        IPath bundlesPath = project.getFullPath().append(
+                                            LiferayWorkspaceUtil.loadConfiguredHomeDir(
+                                                project.getLocation().toOSString() ) );
+
+                                        if( delta.getFullPath().equals( bundlesPath ) )
+                                        {
+                                            try
+                                            {
+                                                File portalBundle =
+                                                    deletedRes.getLocation().toFile().getCanonicalFile();
+
+                                                ServerUtil.deleteRuntimeAndServer( PortalRuntime.ID , portalBundle );
+
+                                                return false;
```

Comment:
Just return true, if we return false it could kill other changes that we need to process.